### PR TITLE
[kube-prometheus-stack] allow setting `GOGC` env configure

### PR DESCRIPTION
**Pull request by dongjiang1989**
#### What this PR does / why we need it
optimize memory usage of secret/configmap informers with `GOGC` env configure.

ref: https://github.com/prometheus-operator/prometheus-operator/pull/5993

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/values.yaml
Comment (dongjiang1989) [inline charts/kube-prometheus-stack/values.yaml:2323]:
```suggestion
  ## Operator Environment
```

**Comment by dongjiang1989** [comment]
@GMartinez-Sisti  PTAL about [shellcheck lint](https://github.com/prometheus-community/helm-charts/pull/4058/commits/a3370970e3a8dc2587884414c5fa3b4fb4164dc6)

**Comment by GMartinez-Sisti** [comment]
> @GMartinez-Sisti PTAL about [shellcheck lint](https://github.com/prometheus-community/helm-charts/pull/4058/commits/a3370970e3a8dc2587884414c5fa3b4fb4164dc6)

Hi @dongjiang1989, this will fix it https://github.com/prometheus-community/helm-charts/pull/4073. Waiting for approval.

**Comment by dongjiang1989** [comment]
> > @GMartinez-Sisti PTAL about [shellcheck lint](https://github.com/prometheus-community/helm-charts/pull/4058/commits/a3370970e3a8dc2587884414c5fa3b4fb4164dc6)
> 
> 
> 
> Hi @dongjiang1989, this will fix it https://github.com/prometheus-community/helm-charts/pull/4073. Waiting for approval.

Thanks for you review. Got it

**Comment by dongjiang1989** [comment]
@GMartinez-Sisti  PTAL

File: charts/kube-prometheus-stack/values.yaml
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/values.yaml:2545]: Why are you setting it to 30 when 100 is the default ?

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/values.yaml:2545]: I read the PR you linked, makes a lot of sense now :D

**Review by QuentinBisson — APPROVED** [review]

File: charts/kube-prometheus-stack/values.yaml
Comment (dongjiang1989) [reply charts/kube-prometheus-stack/values.yaml:2545]:
```
The latter. My observation is that the memory usage increases a lot initially when the operator populates its local cache but it's transient because most of the data is discarded by the transform func and reclaiming more aggressively than the default is beneficial in this case.
```
From the discussion between @philipgough  and @simonpasquier. ref: https://github.com/prometheus-operator/prometheus-operator/pull/5993/files/2c67ca6fc45549a588bd39d362b9bad167b0c025#r1393911325

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/values.yaml:2545]: Thank you for the contribution @dongjiang1989 :)

**Pull request merged**
